### DISCUSSION TITLE
Exporting geometry with decent float32 precision, IfcConvert not doing what I expect.

### DISCUSSION THREAD

**Discussion by peter-slack** [IfcOpenShell/IfcOpenShell#5858] [Q&A] [answered]
Hi, im a beginner with IfcOpenShell so I may be missing something obvious.

My intent is to use IfcConvert.exe to export some ifc files as a set of objects, ideally with their bounding boxes around the origin for maximum floating point precision.

I see some advice:
```
If your IFC uses large map coordinates and your desired format cannot handle it:
IfcConvert --center-model /path/to/input.ifc /path/to/output.glb
# Alternatively:
IfcConvert --center-model-geometry /path/to/input.ifc /path/to/output.glb
# Or you can specify a manual offset in X;Y;Z format
IfcConvert --model-offset "10000;10000;0" /path/to/input.ifc /path/to/output.glb
```

but none of these options seem to affect my output in any way. Also the --precision option does not affect anything. So long as I dont use the -j option the resultant obj and dae are identical apart from timestamps.

Currently it seems easier to just export as OBJ with the --use-element-guids option and sort out bounding boxes myself. The precision in the obj looks high but it is hard to tell if the output decimal places are meaningful.

I really have two questions: (a) what is the best way to achieve my goal, and (b) how do I get these commandline options to produce any effect. Im probably just misunderstanding what they are intended to do so not understanding why they do nothing with the IFC files I am working with. I saw some test ifc files in the git source (I built that to see if it behaved differently from the prebuilt 0.8.0) but didnt spot if there were specific tests for these options.

**Comment by aothms** [comment, marked as answer]
These command line arguments were still unimplemented in v0.8 until recently. Can you retry with latest builds https://builds.ifcopenshell.org/

I think model offset needs to be supplied as `--model-offset 10000 10000 0` now.

Pull Requests with updates to the documentation are always appreciated :)

**Comment by peter-slack** [comment]
Hi thanks.. just confirming that I tried with most recent commit and that new format and it worked :)

commit 99c752904a517f9db893bd8866986b3376a2c854 on 13/dec/2024
IfcConvert.exe .\IFC_files\other\freecad-cylinder.ifc .\output\freecad-cylinder_b.obj --model-offset 11 22 33
